tests/unit/models/test_decay.py

```diff
@@ -0,0 +1,136 @@
+
+        # Filter discontinuities
+        unique_indices = np.unique(sol.t, return_index=True)[1]
+
+        return pybop.Dataset(
+            {
+                "Time [s]": sol.t[unique_indices],
+                "y_0": sol["y_0"].data[unique_indices],
+            }
+        )
+
+    @pytest.fixture
+    def test_parameters(self):
+        """Create parameter objects for reuse."""
+        return [
+            pybop.Parameter(param_name, initial_value=val)
+            for param_name, val in EXPONENTIAL_DECAY_PARAMS
+        ]
+
+    def create_pybamm_builder(self, dataset, model_config: dict[str, Any], parameters):
+        """Factory function for creating Pybamm builders."""
+        builder = pybop.builders.Pybamm()
+        builder.set_dataset(dataset)
+        builder.set_simulation(
+            model_config["model"], parameter_values=model_config["parameter_values"]
+        )
+
+        for parameter in parameters:
+            builder.add_parameter(parameter)
+
+        return builder
+
+    def assert_parameter_sensitivity(
+        self, problem, initial_params, test_params, tolerance=RELATIVE_TOLERANCE
+    ):
+        """Reusable assertion for parameter sensitivity."""
+        problem.set_params(initial_params)
+        value1 = problem.run()
+
+        problem.set_params(test_params)
+        value2 = problem.run()
+
+        relative_change = abs((value1 - value2) / value1)
+        assert relative_change > tolerance, (
+            f"Parameter change insufficient: {relative_change}"
+        )
+
+        return value1, value2
+
+    def test_decay_builder(self, dataset, base_model_config, test_parameters):
+        """Test decay model with voltage-based cost functions."""
+        builder = self.create_pybamm_builder(
+            dataset, base_model_config, test_parameters
+        )
+        builder.add_cost(pybop.costs.pybamm.SumSquaredError("y_0", "y_0"))
+        builder.add_cost(pybop.costs.pybamm.MeanAbsoluteError("y_0", "y_0"))
+
+        problem = builder.build()
+
+        # Test parameter sensitivity
+        initial_params = problem.params.get_initial_values()
+
+        value1, value2 = self.assert_parameter_sensitivity(
+            problem, initial_params, TEST_PARAM_VALUES
+        )
+
+        # Test sensitivity computation consistency
+        problem.set_params(initial_params)
+        value1_sens, grad1 = problem.run_with_sensitivities()
+
+        problem.set_params(TEST_PARAM_VALUES)
+        value2_sens, grad2 = problem.run_with_sensitivities()
+
+        # Validate gradient shape and value consistency
+        assert grad1.shape == (len(test_parameters),), (
+            f"Gradient shape mismatch: {grad1.shape}"
+        )
+        assert grad2.shape == (len(test_parameters),), (
+            f"Gradient shape mismatch: {grad2.shape}"
+        )
+
+        np.testing.assert_allclose(
+            value1_sens, value1, atol=ABSOLUTE_TOLERANCE, rtol=RELATIVE_TOLERANCE
```

Comment:
if the model is consistently wrong, this won't be caught. Since its an exponential decay I think we should test against the analytical solutions here, for the value and the sensitivity.